### DISCUSSION TITLE
Data management with React Hooks

### DISCUSSION THREAD

**Discussion by ryanwelcher** [WordPress/developer-blog-content#53] [Topic Ideas]
Now that #25 is published, it would be great to have a similar post that explains how to do the same using Hooks instead of Higher Order Components. HOCs are an older way to approach adding functionality to components. While they are not incorrect in any way, using a hook version is often simpler to implement and understand. 

Having a post that shows how to use the hooks and explains the differences would be very helpful

**Comment by bph** [comment]
That is so interesting. How would you approach it? Is there some documentation as a jumping off point? Or some use cases?

**Reply by ryanwelcher** [reply]
There is actually a brief explanation of the Hook equivalents at the end of the article that is published now. Perhaps, we can take that great intro, build on it and show some other ones to be able to retrieve data.

Maybe this article is more along the lines of "XX Hooks you need to know for WordPress development"?

**Comment by bph** [comment]
After a more beginners explanation, this could also be a follow-up on the Dev Note: [Simplified data access with new React hooks in WordPress 6.1](https://make.wordpress.org/core/2022/10/11/simplified-data-access-with-new-react-hooks-in-wordpress-6-1/) 
As well as the updated documentation of the Hooks reference
[Hooks Reference](https://developer.wordpress.org/block-editor/reference-guides/filters/)
